src/haddock/modules/analysis/alascan/scan.py

```diff
@@ -429,175 +434,325 @@ def create_alascan_plots(clt_alascan, scan_residue, offline = False):
+            List of mutation jobs if library_mode=False, None if library_mode=True
+        """
+        try:
+            # Calculate native scores
+            sc_dir = f"haddock3-score-{self.model_id}-{os.getpid()}"
+            try:
+                native_scores = calc_score(self.model_path, run_dir=sc_dir, outputpdb=False)
+            finally:
+                if os.path.exists(sc_dir):
+                    shutil.rmtree(sc_dir)
+            
+            # Load coordinates
+            atoms = get_atoms(self.model_path)
+            coords, chain_ranges = load_coords(self.model_path, atoms, add_resname=True)
             
-            # check if the user wants to mutate only some residues
+            # Determine target residues: get interface, then apply user filers, if given 
+            # Get all interface residues        
+            cutoff = self.params.get("int_cutoff", 5.0)
+            interface = CAPRI.identify_interface(self.model_path, cutoff=cutoff)
+            
+            # get user_chains for the chek down the line
+            user_chains = self.params.get("chains", [])
+
+            # if user defined target residues, check they are in the interface
             if self.filter_resdic != {'_': []}:
-                interface = {}
+                filtered_interface = {}
                 for chain in self.filter_resdic:
-                    if chain in chain_ranges:
-                        chain_aas = [aa[1] for aa in coords if aa[0] == chain]
-                        unique_aas = list(set(chain_aas))
-                        # the interface here is the intersection of the
-                        # residues in filter_resdic and the residues actually 
-                        # present in the model
-                        interface[chain] = [
-                            res for res in self.filter_resdic[chain]
-                            if res in unique_aas
-                            ]
-            else:
-                interface = CAPRI.identify_interface(
-                    native.rel_path,
-                    cutoff=self.int_cutoff
-                    )
-                # in case the user wants to scan only some chains (this is
-                # superseded by filter_resdic)
-                if self.chains != []:
-                    interface = {
-                        chain: interface[chain] for chain in self.chains
-                        if chain in interface
-                        }
-            
+                    if chain in interface:
+                        user_res_valid = []
+                        for res in self.filter_resdic[chain]:
+                            if res in interface[chain]:
+                                user_res_valid.append(res)
+                        if user_res_valid:
+                            user_res_valid_unique = list(set(user_res_valid))
+                            filtered_interface[chain] = user_res_valid_unique
```

Comment:
Here you are searching for the intersection of two sets.

what about:
```suggestion
                        # Search for the intersection of user queried residues and interface residues
                        user_res_valid = list(set(self.filter_resdic[chain]).intersection(set(interface[chain])))
                        # If at least one residue must be analyzed, add it to residues to be scanned
                        if len(user_res_valid) >= 1:
                            filtered_interface[chain] = user_res_valid
```
?

Reply:
yes, great suggestion!